### PR TITLE
feature(views): support for extra variables in elgg_view_icon function

### PR DESCRIPTION
**Pull request by jdalsem**
allows you to do this

``` php
elgg_view_icon("info", "float-alt", array("title" => elgg_echo("tooltip:text")));
```

### CODE REVIEW

File: engine/lib/views.php [hunk 1]
```diff
@@ -1367,18 +1367,36 @@ function elgg_view_list_item($item, array $vars = array()) {
  *
  * @param string $name  The specific icon to display
  * @param string $class Additional class: float, float-alt, or custom class
+ * @param array  $vars  Variables to pass to the view.
  *
  * @return string The html for displaying an icon
  */
-function elgg_view_icon($name, $class = '') {
+function elgg_view_icon($name, $class = '', $vars = array()) {
+	if (!is_array($vars)) {
```
Comment (mrclay) [inline engine/lib/views.php]: Just type-hint the argument

File: engine/lib/views.php [hunk 2]
```diff
@@ -1367,18 +1367,36 @@ function elgg_view_list_item($item, array $vars = array()) {
  *
  * @param string $name  The specific icon to display
  * @param string $class Additional class: float, float-alt, or custom class
+ * @param array  $vars  Variables to pass to the view.
  *
  * @return string The html for displaying an icon
  */
-function elgg_view_icon($name, $class = '') {
+function elgg_view_icon($name, $class = '', $vars = array()) {
```
Comment (mrclay) [inline engine/lib/views.php]: Any reason we shouldn't just replace $class with $vars?

Comment (mrclay) [inline engine/lib/views.php]: If is string, `$vars = ['class' => $vars]`

Comment (jdalsem) [inline engine/lib/views.php]: same as done for getIconURL()? mixed param? fine by me, but can't type hint then anymore :)

**Comment by jdalsem** [comment]
changed to a mixed param

File: engine/lib/views.php [hunk 3]
```diff
@@ -1366,19 +1366,35 @@ function elgg_view_list_item($item, array $vars = array()) {
  * Shorthand for <span class="elgg-icon elgg-icon-$name"></span>
  *
  * @param string $name  The specific icon to display
- * @param string $class Additional class: float, float-alt, or custom class
+ * @param mixed  $vars  Class string or variables to pass to the icon view.
```
Comment (mrclay) [inline engine/lib/views.php]: We should say how to set class and list the example values we had before.

File: engine/lib/views.php [hunk 4]
```diff
@@ -1366,19 +1366,35 @@ function elgg_view_list_item($item, array $vars = array()) {
...
+		$vars = array('class' => 'float');
 	}
 	
-	$icon_class = array("elgg-icon-$name" , $class);
+	if (is_string($vars)) {
+		$vars = array('class' => $vars);
+	}
+	
+	if (!is_array($vars)) {
+		$vars = array();
```
Comment (mrclay) [inline engine/lib/views.php]: at this point we would throw InvalidArgumentException.

**Comment by jdalsem** [comment]
updated description and now throwing an exception if !is_array

**Comment by mrclay** [comment]
LGTM

**Comment by jdalsem** [comment]
needed to add `if (empty($vars)) { $vars = array(); }` because there are occurences of `elgg_view_icon('icon_name', NULL)` that would trigger the argumentexception.

**Comment by jdalsem** [comment]
Issue is originating in https://github.com/Elgg/Elgg/blob/1.10/views/default/output/tags.php#L66 (`$icon_class` = null). Should i fix it also?

File: engine/lib/views.php [hunk 5]
```diff
@@ -1365,20 +1365,43 @@ function elgg_view_list_item($item, array $vars = array()) {
...
+ * @param mixed  $vars The additional classname as a string ('float', 'float-alt' or a custom class) 
+ *                     or an array of variables (array('class' => 'float')) to pass to the icon view.
  *
  * @return string The html for displaying an icon
+ * @throws InvalidArgumentException
  */
-function elgg_view_icon($name, $class = '') {
-	if ($class === true) {
+function elgg_view_icon($name, $vars = array()) {
+	if (empty($vars)) {
```
Comment (mrclay) [inline engine/lib/views.php]: Seems fine to me. I see no urgency to remove this in 2.0

Comment (jdalsem) [inline engine/lib/views.php]: i removed the comment

**Comment by mrclay** [comment]
LGTM

**Pull request merged**